packages/context/src/interceptor.ts

```diff
@@ -315,3 +315,36 @@ function invokeInterceptors(
   };
   return next();
 }
+
+/**
+ * A proxy handler that applies interceptors
+ *
+ * See https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Proxy
+ */
+export class InterceptorHandler<T extends object> implements ProxyHandler<T> {
```

Comment:
Please improve the name to describe the intent (apply/invoke interceptors), not implementation details (interceptor-related handler). E.g. `InterceptorInvoker` instead of `InterceptorHandler`.